nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,336 @@
+      DeviceAuthorizationFlow = {
+        Provider = NETBIRD_AUTH_DEVICE_AUTH_PROVIDER;
+        ProviderConfig = {
+          Audience = NETBIRD_AUTH_DEVICE_AUTH_AUDIENCE;
+          Domain = NETBIRD_AUTH0_DOMAIN;
+          ClientID = NETBIRD_AUTH_DEVICE_AUTH_CLIENT_ID;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          DeviceAuthEndpoint = NETBIRD_AUTH_DEVICE_AUTH_ENDPOINT;
+          Scope = builtins.concatStringsSep " " NETBIRD_AUTH_DEVICE_AUTH_SCOPE;
+          UseIDToken = NETBIRD_AUTH_DEVICE_AUTH_USE_ID_TOKEN;
+        };
+      };
+      PKCEAuthorizationFlow = {
+        ProviderConfig = {
+          Audience = NETBIRD_AUTH_AUDIENCE;
+          ClientID = NETBIRD_AUTH_CLIENT_ID;
+          ClientSecret = NETBIRD_AUTH_CLIENT_SECRET;
+          AuthorizationEndpoint = NETBIRD_AUTH_PKCE_AUTHORIZATION_ENDPOINT;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          Scope = builtins.concatStringsSep " " NETBIRD_AUTH_SUPPORTED_SCOPES;
+          RedirectURLs = NETBIRD_AUTH_PKCE_REDIRECT_URLS;
+          UseIDToken = NETBIRD_AUTH_PKCE_USE_ID_TOKEN;
+        };
+      };
+    };
+
+    networking.firewall.allowedTCPPorts = [ 80 443 33073 10000 ];
+
+    services.nginx.virtualHosts = {
+      ${cfg.settings.NETBIRD_DOMAIN} = {
+        forceSSL = true;
+        enableACME = true;
+
+        locations."/" = {
+          root = "${stateDir}/web-ui/";
+          tryFiles = "$uri /index.html";
+        };
+
+        locations."/signalexchange.SignalExchange/".extraConfig = ''
+          grpc_pass grpc://localhost:${builtins.toString cfg.ports.signal};
+          # grpc_ssl_verify off;
+          grpc_read_timeout 1d;
+          grpc_send_timeout 1d;
+          grpc_socket_keepalive on;
+        '';
+
+        locations."/api".proxyPass = "http://localhost:${builtins.toString cfg.ports.management}";
+
+        locations."/management.ManagementService/".extraConfig = ''
+          grpc_pass grpc://localhost:${builtins.toString cfg.ports.management};
+          # grpc_ssl_verify off;
+          grpc_read_timeout 1d;
+          grpc_send_timeout 1d;
+          grpc_socket_keepalive on;
+        '';
+      };
+    };
+
+    systemd.services.netbird-setup = {
+      wantedBy = [ "netbird-management.service" "netbird-signal.service" "multi-user.target" ];
+      serviceConfig = {
+        Type = "oneshot";
+        RuntimeDirectory = "netbird-mgmt";
+        StateDirectory = "netbird-mgmt";
+        WorkingDirectory = stateDir;
+        EnvironmentFile = [ settingsFile ];
+      };
+      unitConfig = {
+        StartLimitInterval = 5;
+        StartLimitBurst = 10;
+      };
+
+      path = with pkgs; [ coreutils findutils gettext gnused ];
+
+      script = ''
+        cp ${managementFile} ${stateDir}/config.json
+        cat ${stateDir}/config.json
+
+        mkdir -p ${stateDir}/data
+
+        rm -rf ${stateDir}/web-ui
+        mkdir -p ${stateDir}/web-ui
+        cp -R ${pkgs.netbird-dashboard}/* ${stateDir}/web-ui
```

Comment:
Do you have to consistently recreate it or could something like rsync with appropriate flags could update the new files?

Reply:
(explain why this is needed also)

Reply:
It is needed because some configuration values, e.g. the auth endpoint are hardcoded in the web-ui and replaced from env varibles with envsubst. I suppose that we could also pass those variables to the dashboard package and set them at build time.

Reply:
It's quite unfortunate that they require envsubst instead of accessing the environment variables directly... I don't like copying each time but it seems like this is currently necessary - unless upstream changes the process.

On that note: In https://github.com/netbirdio/dashboard/issues/242 one of the maintainers (?) noted that they would accept contributions to make this process easier to host without docker. I imagine that a simple PR to upstream that replaces this envsubst by just accessing `process.env.SomeVar` in their config.ts would solve this for us and for them.

Reply:
I mean, we can also do it ourselves with substituteAll while building the dashboard, but we need to decide if we want to rebuild the dashboard as a whole every time an option change, which would take some time

Reply:
IMO an upstream solution seems to be aligned with their goals anyway, so only if they refuse I would start resorting to alternatives.